<a href="https://colab.research.google.com/github/rlcjuliaguimaraes-collab/aurora-siger/blob/main/aurora_siger_completo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 🚀 Relatório Operacional de Pré-Decolagem — Aurora Siger

**Missão:** Ignition Zero — O Início da Aurora  
**Objetivo:** Validar parâmetros técnicos para autorizar ou abortar o lançamento da nave Aurora Siger.

---

## 1. Organização e Descrição da Telemetria

A telemetria é o conjunto de dados enviados automaticamente pelos sistemas embarcados da nave para monitoramento em tempo real. Durante a pré-decolagem, os seguintes parâmetros são monitorados:

| Parâmetro | Valor | Unidade | Faixa Segura | Significado |
|---|---|---|---|---|
| Temperatura interna | 22.5 | °C | 18°C – 30°C | Condição habitável e segura para os sistemas eletrônicos |
| Temperatura externa | -60.0 | °C | Esperado em altitude | Condição esperada fora da nave; isolamento deve ser verificado |
| Integridade estrutural | 1 | binário (0/1) | 1 = OK | Confirma que a estrutura da nave está íntegra |
| Nível de energia | 87.3 | % | ≥ 80% | Carga atual das baterias; margem mínima para decolagem segura |
| Pressão do tanque | 350.0 | kPa | 300 – 400 kPa | Pressão dos propelentes; fora do range causa falha de ignição |
| Módulo de propulsão | 1 | binário (0/1) | 1 = ativo | Sistema responsável pelo impulso da nave |
| Módulo de comunicação | 1 | binário (0/1) | 1 = ativo | Sistema de transmissão de dados com a base terrestre |
| Módulo de navegação | 1 | binário (0/1) | 1 = ativo | Sistema de orientação e trajetória da nave |

> ⚠️ **Atenção:** O nível de energia (87.3%) está a apenas 7.3 pontos percentuais do limite mínimo de 80%. Monitoramento contínuo recomendado durante a contagem regressiva.

In [ ]:
# Dados de telemetria da nave Aurora Siger
telemetria = {
    "temperatura_interna": 22.5,   # °C — faixa segura: 18 a 30
    "temperatura_externa": -60.0,  # °C — esperado em altitude de cruzeiro
    "integridade_estrutural": 1,   # 1 = OK, 0 = falha
    "nivel_energia": 87.3,         # % — mínimo: 80%
    "pressao_tanque": 350.0,       # kPa — faixa segura: 300 a 400
    "modulo_propulsao": 1,         # 1 = ativo, 0 = falha
    "modulo_comunicacao": 1,       # 1 = ativo, 0 = falha
    "modulo_navegacao": 1,         # 1 = ativo, 0 = falha
}

print("=" * 50)
print("  TELEMETRIA — NAVE AURORA SIGER")
print("=" * 50)
for chave, valor in telemetria.items():
    print(f"  {chave:<25} → {valor}")
print("=" * 50)

  TELEMETRIA — NAVE AURORA SIGER
  temperatura_interna       → 22.5
  temperatura_externa       → -60.0
  integridade_estrutural    → 1
  nivel_energia             → 87.3
  pressao_tanque            → 350.0
  modulo_propulsao          → 1
  modulo_comunicacao        → 1
  modulo_navegacao          → 1


---
## 2. Algoritmo de Verificação

### 2.1 Pseudocódigo

```
INÍCIO verificar_decolagem(telemetria):
  falhas ← lista vazia

  SE temperatura_interna < 18 OU temperatura_interna > 30:
      adicionar "Temperatura interna fora do range" em falhas

  SE integridade_estrutural ≠ 1:
      adicionar "Falha na integridade estrutural" em falhas

  SE nivel_energia < 80:
      adicionar "Energia abaixo do mínimo" em falhas

  SE pressao_tanque < 300 OU pressao_tanque > 400:
      adicionar "Pressão do tanque fora do range" em falhas

  PARA CADA modulo EM [propulsao, comunicacao, navegacao]:
      SE modulo ≠ 1:
          adicionar "Módulo com falha: " + modulo em falhas

  SE falhas está vazia:
      IMPRIMIR "✅ PRONTO PARA DECOLAR"
  SENÃO:
      IMPRIMIR "❌ DECOLAGEM ABORTADA"
      PARA CADA falha EM falhas:
          IMPRIMIR "→ " + falha

FIM
```

### 2.2 Fluxo lógico

O algoritmo percorre 5 verificações em sequência:
1. Temperatura interna (18°C ≤ T ≤ 30°C)
2. Integridade estrutural (deve ser 1)
3. Nível de energia (deve ser ≥ 80%)
4. Pressão do tanque (300 ≤ P ≤ 400 kPa)
5. Status dos três módulos críticos (todos devem ser 1)

Se **todas** as condições forem satisfeitas → **PRONTO PARA DECOLAR**  
Se **qualquer** condição falhar → **DECOLAGEM ABORTADA** + lista de falhas

---
## 3. Script em Python — Verificação Automatizada

In [ ]:
def verificar_decolagem(t):
    """
    Verifica os parâmetros de telemetria da nave Aurora Siger.
    Retorna 'PRONTO PARA DECOLAR' ou 'DECOLAGEM ABORTADA' com lista de falhas.

    Parâmetros verificados:
    - Temperatura interna: 18°C a 30°C
    - Integridade estrutural: deve ser 1 (OK)
    - Nível de energia: mínimo 80%
    - Pressão do tanque: 300 a 400 kPa
    - Módulos críticos (propulsão, comunicação, navegação): devem ser 1
    """
    falhas = []

    # Verificação 1: Temperatura interna
    if not (18 <= t["temperatura_interna"] <= 30):
        falhas.append(f"Temperatura interna fora do range (valor: {t['temperatura_interna']}°C)")

    # Verificação 2: Integridade estrutural
    if t["integridade_estrutural"] != 1:
        falhas.append("Falha na integridade estrutural")

    # Verificação 3: Nível de energia
    if t["nivel_energia"] < 80:
        falhas.append(f"Energia abaixo do mínimo (valor: {t['nivel_energia']}%)")

    # Verificação 4: Pressão do tanque
    if not (300 <= t["pressao_tanque"] <= 400):
        falhas.append(f"Pressão do tanque fora do range (valor: {t['pressao_tanque']} kPa)")

    # Verificação 5: Módulos críticos
    for modulo in ["modulo_propulsao", "modulo_comunicacao", "modulo_navegacao"]:
        if t[modulo] != 1:
            falhas.append(f"Módulo com falha: {modulo}")

    # Resultado final
    print("=" * 50)
    print("  RESULTADO DA VERIFICAÇÃO PRÉ-DECOLAGEM")
    print("=" * 50)

    if not falhas:
        print("  ✅ STATUS: PRONTO PARA DECOLAR")
        print("  Todos os parâmetros dentro das faixas seguras.")
    else:
        print("  ❌ STATUS: DECOLAGEM ABORTADA")
        print(f"  {len(falhas)} falha(s) detectada(s):")
        for f in falhas:
            print(f"    → {f}")

    print("=" * 50)

# Executar verificação
verificar_decolagem(telemetria)

  RESULTADO DA VERIFICAÇÃO PRÉ-DECOLAGEM
  ✅ STATUS: PRONTO PARA DECOLAR
  Todos os parâmetros dentro das faixas seguras.


### 3.1 Teste com cenário de falha

Para demonstrar o funcionamento do algoritmo em caso de parâmetros fora das faixas:

In [ ]:
# Cenário de teste: simulando falhas para validar o algoritmo
telemetria_com_falhas = {
    "temperatura_interna": 35.0,   # FORA DO RANGE (acima de 30°C)
    "temperatura_externa": -60.0,
    "integridade_estrutural": 1,
    "nivel_energia": 72.0,          # FORA DO RANGE (abaixo de 80%)
    "pressao_tanque": 350.0,
    "modulo_propulsao": 1,
    "modulo_comunicacao": 0,         # FALHA no módulo de comunicação
    "modulo_navegacao": 1,
}

verificar_decolagem(telemetria_com_falhas)

  RESULTADO DA VERIFICAÇÃO PRÉ-DECOLAGEM
  ❌ STATUS: DECOLAGEM ABORTADA
  3 falha(s) detectada(s):
    → Temperatura interna fora do range (valor: 35.0°C)
    → Energia abaixo do mínimo (valor: 72.0%)
    → Módulo com falha: modulo_comunicacao


---
## 4. Análise Energética

In [ ]:
# Parâmetros energéticos da nave
capacidade_total_kwh = 500        # Capacidade total das baterias (kWh)
carga_atual_pct      = telemetria["nivel_energia"]  # 87.3%
consumo_decolagem_kwh = 120       # Consumo estimado durante a decolagem (kWh)
perda_estimada_pct   = 5          # Perdas por calor, conversão e distribuição (%)

# Cálculos
energia_disponivel  = capacidade_total_kwh * (carga_atual_pct / 100)
energia_apos_perda  = energia_disponivel * (1 - perda_estimada_pct / 100)
autonomia_restante  = energia_apos_perda - consumo_decolagem_kwh
autonomia_pct       = (autonomia_restante / capacidade_total_kwh) * 100

# Relatório energético
print("=" * 50)
print("  ANÁLISE ENERGÉTICA — AURORA SIGER")
print("=" * 50)
print(f"  Capacidade total:          {capacidade_total_kwh:.1f} kWh")
print(f"  Carga atual:               {carga_atual_pct:.1f}%")
print(f"  Energia disponível:        {energia_disponivel:.1f} kWh")
print(f"  Perdas estimadas ({perda_estimada_pct}%):    -{energia_disponivel - energia_apos_perda:.1f} kWh")
print(f"  Energia após perdas:       {energia_apos_perda:.1f} kWh")
print(f"  Consumo na decolagem:      -{consumo_decolagem_kwh:.1f} kWh")
print("-" * 50)
print(f"  Reserva pós-decolagem:     {autonomia_restante:.1f} kWh ({autonomia_pct:.1f}%)")
print("=" * 50)

# Avaliação da margem
if autonomia_restante > 200:
    print("  ✅ Margem energética adequada para a missão.")
elif autonomia_restante > 100:
    print("  ⚠️  Margem energética moderada — monitorar consumo.")
else:
    print("  ❌ Margem energética crítica — risco de falha de missão.")

  ANÁLISE ENERGÉTICA — AURORA SIGER
  Capacidade total:          500.0 kWh
  Carga atual:               87.3%
  Energia disponível:        436.5 kWh
  Perdas estimadas (5%):    -21.8 kWh
  Energia após perdas:       414.7 kWh
  Consumo na decolagem:      -120.0 kWh
--------------------------------------------------
  Reserva pós-decolagem:     294.7 kWh (58.9%)
  ✅ Margem energética adequada para a missão.


---
## 5. Análise Assistida por IA

A análise dos dados de telemetria foi realizada com auxílio de IA (Claude — Anthropic), solicitando classificação dos parâmetros, identificação de anomalias e sugestões de risco.

### 5.1 Classificação dos dados

| Parâmetro | Valor | Faixa Segura | Classificação |
|---|---|---|---|
| Temperatura interna | 22.5°C | 18°C – 30°C | ✅ Normal |
| Temperatura externa | -60°C | Esperado em altitude | ✅ Normal |
| Integridade estrutural | 1 | 1 | ✅ Normal |
| Nível de energia | 87.3% | ≥ 80% | ⚠️ Atenção |
| Pressão do tanque | 350 kPa | 300–400 kPa | ✅ Normal |
| Módulo propulsão | 1 | 1 | ✅ Normal |
| Módulo comunicação | 1 | 1 | ✅ Normal |
| Módulo navegação | 1 | 1 | ✅ Normal |

### 5.2 Anomalias identificadas

Nenhum parâmetro está fora da faixa segura, portanto não há anomalias críticas. A IA sinalizou um ponto de atenção: o nível de energia em 87.3% está a apenas 7.3 pontos percentuais do limite mínimo de 80%. Qualquer consumo inesperado antes da decolagem — como ativação de sistemas auxiliares ou uma falha de curta duração — poderia derrubar esse valor abaixo do mínimo e forçar o abort.

### 5.3 Sugestões de risco

**Risco 1 — Margem energética estreita** | Grau: Baixo-Médio  
O nível de 87.3% é funcional, mas recomenda-se monitoramento contínuo durante a contagem regressiva. Caso desça abaixo de 83%, sugere-se acionar protocolo de conservação energética, desligando sistemas não-críticos.

**Risco 2 — Temperatura externa** | Grau: Informativo  
O valor de -60°C é esperado para a altitude de cruzeiro e não representa falha. Porém os sistemas de isolamento térmico devem ser verificados para garantir que a temperatura interna se mantenha estável durante a decolagem.

**Risco 3 — Ausência de redundância registrada** | Grau: Informativo  
Os três módulos críticos estão ativos, mas o relatório não registra sistemas de backup. Em missões reais, recomenda-se que cada módulo crítico tenha redundância verificada antes da autorização de decolagem.

---
## 6. Reflexão Crítica

### 6.1 Ética e Responsabilidade

A construção de sistemas computacionais aplicados a missões espaciais exige um nível de responsabilidade ética que vai muito além da correção técnica do código. Cada linha de algoritmo escrita para verificar parâmetros de decolagem representa uma decisão que pode impactar diretamente vidas humanas e recursos de valor incalculável. Nesse contexto, o desenvolvedor não é apenas um programador — é um agente responsável por garantir que sistemas autônomos tomem decisões seguras, transparentes e auditáveis.

A missão Aurora Siger ilustra bem esse princípio. O algoritmo de verificação foi construído com faixas seguras predefinidas e critérios objetivos, o que garante rastreabilidade: é possível saber exatamente por que uma decolagem foi liberada ou abortada. Essa transparência é um requisito ético fundamental em sistemas críticos. Quando uma IA ou algoritmo toma uma decisão sem que seus critérios possam ser explicados ou revisados, surgem riscos sérios de falhas silenciosas — erros que ninguém percebe até que seja tarde demais.

Além disso, o uso de inteligência artificial como suporte analítico reforça a necessidade de supervisão humana. A IA pode identificar padrões e sugerir riscos, mas a decisão final deve sempre passar por uma camada de validação humana consciente.

### 6.2 Impacto Social da Exploração Espacial

A exploração espacial é frequentemente celebrada como um triunfo da humanidade, e de fato representa conquistas científicas extraordinárias. No entanto, é necessário olhar também para suas dimensões sociais e políticas. Os investimentos bilionários direcionados a programas espaciais levantam questionamentos legítimos sobre prioridades: enquanto missões interplanetárias são financiadas, bilhões de pessoas ainda carecem de acesso à água potável, educação de qualidade e infraestrutura básica.

Esse debate não significa que a exploração espacial deva ser abandonada, mas que ela precisa ser conduzida com consciência social. Programas como o Artemis e iniciativas privadas como a SpaceX têm o potencial de gerar tecnologias que beneficiem a sociedade como um todo — desde avanços em energia solar até novos materiais e sistemas de comunicação. A missão Aurora Siger, ao simular um ecossistema aeroespacial real, também convida o estudante a refletir sobre quem tem acesso a esse tipo de tecnologia e conhecimento.

### 6.3 Sustentabilidade Tecnológica

A sustentabilidade na exploração espacial não se limita à questão ambiental terrestre, embora ela seja parte fundamental da discussão. O lançamento de foguetes consome enormes quantidades de combustível e gera emissões significativas. A busca por propulsores mais eficientes, o uso de combustíveis menos poluentes e o reaproveitamento de componentes — como já praticado pela SpaceX com o pouso dos boosters — são avanços concretos em direção a uma exploração mais sustentável.

No contexto do projeto, a análise energética da nave Aurora demonstra na prática o princípio da eficiência como imperativo técnico e sustentável. Com uma reserva de 294.7 kWh após a decolagem, cada decisão computacional que consome energia desnecessária compromete a autonomia da missão. Algoritmos eficientes não são apenas uma boa prática de programação — são uma necessidade de sobrevivência no espaço, onde energia é um recurso finito e precioso.

Por fim, a sustentabilidade tecnológica envolve também a responsabilidade com o ambiente espacial. A proliferação de satélites e detritos orbitais já representa um risco real para missões futuras. O conceito de ESG — ambiental, social e de governança — precisa ser incorporado não apenas nas empresas terrestres, mas em toda cadeia de decisão da exploração espacial.

---
## Referências

- ESA. *European Space Agency*. Disponível em: https://www.esa.int/
- NASA. *Autonomous Systems*. Disponível em: https://www.nasa.gov/centers-and-facilities/armstrong/autonomous-systems/
- SPACEX. *Starship*. Disponível em: https://www.spacex.com/vehicles/starship